flask_resty/authentication.py

```diff
@@ -62,3 +65,45 @@ class NoOpAuthentication:
 
     def authenticate_request(self):
         pass
+
+
+# -----------------------------------------------------------------------------
+
+
+class HeaderAuthenticationBase(AuthenticationBase):
+    """Base class for Authentication components that get their credentials from
+    the Authorization request header. The Authorization header has the form:
+
+        Authorization: <scheme> <credentials>
+    """
+
+    #: Corresponds to the <scheme> in the Authorization request header.
+    header_scheme = None
+
+    #: A fallback query parameter. The value of this query parameter will be
+    #: used as credentials if the Authorization request header is missing.
+    credentials_arg = None
+
+    def get_request_credentials(self):
+        authorization = flask.request.headers.get("Authorization")
+
+        if authorization is None:
+            if self.credentials_arg is not None:
+                return flask.request.args.get(self.credentials_arg)
+            raise ApiError(401, {"code": "invalid_authorization.missing"})
+
+        return self.get_credentials_from_authorization(authorization)
+
+    def get_credentials_from_authorization(self, authorization):
+        if self.header_scheme is None:
```

Comment:
could just default this to Bearer. the other schemes all have handshakes and challenges that were not implementing